rel/expr.go

```diff
@@ -32,21 +32,50 @@ func (e ExprScanner) Source() parser.Scanner {
 	return e.Src
 }
 
-func wrapContext(err error, expr Expr) error {
-	return fmt.Errorf("%s\n%s", err.Error(), expr.Source().Context(parser.DefaultLimit))
+type ContextErr struct {
+	err    error
+	source parser.Scanner
+	scope  Scope
+}
+
+func (c ContextErr) Error() string {
+	return fmt.Sprintf("%s\n%s", c.err.Error(), c.source.Context(parser.DefaultLimit))
+}
+
+// GetLastScope gets the scope nearest to the error
+func (c ContextErr) GetLastScope() Scope {
+	ctxErr := c
+	for {
+		var isContextErr bool
+		var currentErr ContextErr
+		if currentErr, isContextErr = ctxErr.err.(ContextErr); !isContextErr {
+			return ctxErr.scope
+		}
+		ctxErr = currentErr
+	}
+}
+
+func wrapContext(err error, expr Expr, scope Scope) error {
+	// removes stdlib
+	scope = scope.Without(".")
+	if _, isContextErr := err.(ContextErr); isContextErr {
+		return ContextErr{err, expr.Source(), EmptyScope}
```

Comment:
Does it need to be removed? Preserving the chain of scopes might help when implementing stack frame management in the debugger (show me the local names in the function that called me).